### DISCUSSION TITLE
Use of spacy

### DISCUSSION THREAD

**Discussion by standenman** [medspacy/medspacy#244] [General]
I am having trouble understanding what Medspacy offers above just using.  I know it is designed for medical and clinical texts - that is why I am look at it.  But my output is  worthless from a medical point of view.  Diseases missed, symptoms missed.  I don't get it.
![Screenshot (90)](https://github.com/medspacy/medspacy/assets/17820082/7cf577d4-42ea-474f-8e66-23355ac6bf89)

**Comment by abchapman93** [comment]
Hi @standenman ,

Referring back to my comment to your [previous discussion](https://github.com/medspacy/medspacy/discussions/235#discussioncomment-6019458), here are a few points worth considering:

medspaCy is not designed to be an out-of-the-box, general clinical NLP model. Instead, it's a toolkit of modularized components that can be used to build custom-designed pipelines for specific applications. As such, a lot of the functionality in medspaCy has focused around rule-based NLP, with the expectation that the user will be writing the rules to extract certain attributes. For specific examples of this, you could refer to some of my personal repositories with pipelines for [Covid-19 surveillance](https://github.com/abchapman93/VA_COVID-19_NLP_BSV) and [pneumonia](https://github.com/abchapman93/medspacy_pneumonia). We also have extensive [notebooks](https://github.com/medspacy/medspacy/tree/master/notebooks) documenting the main expected use cases of medspaCy.

medspaCy is also compatible with pretrained statistical models like the kind you might be looking for, but we do not fit or provide these models. scispaCy may be useful to you, or you can search for other models, perhaps hosted by huggingface. You might also find the QuickUMLS component useful, as this can be used to extract general medical concepts without writing extensive rules. I'll again refer to the [notebooks](https://github.com/medspacy/medspacy/blob/master/notebooks/11a-QuickUMLS_Extraction_Defaults.ipynb) on how to use this component. 

I hope this information is helpful.

**Reply by standenman** [reply]
Thanks very much!

**Comment by abchapman93** [comment]
I'll also note that the model you are using `en_core_web_sm` is a spaCy model trained on general purpose English text, so the NER component will not extract medical concepts.

**Reply by standenman** [reply]
Oh what should I be using?  I could not get Scispacy installed because of issues with nmlib, "wheels" ad naseum.

**Reply by turbosheep** [reply]
Based on your previous description of the project you're working on, I would highly suggest getting QuickUMLS set up for your project.
 The instructions can be seen here: [https://github.com/medspacy/QuickUMLS#installation](https://github.com/medspacy/QuickUMLS#installation) 
 
Since you already have medspacy installed and quickumls now installs with a medspacy installation on all platforms, step 2 is not required, but 1 and 3 will be necessary. QuickUMLS captures a broad range of medical concepts in text using an edit-distance based matching technique, but it sacrifices accuracy to do so without training data.

There are unfortunately very few machine-learning based models (like `en_core_web_sm` that you were using) that will capture the broad information you describe in your previous question. The ones that I know of are proprietary, such as [Amazon Medical Comprehend](https://aws.amazon.com/comprehend/medical/) which will require you send your data to amazon for processing and the output is not immediately compatible with spacy or any other spacy-based libraries like medspacy. These will most likely be more accurate than quickUMLS but may not be acceptable for your project for a variety of reasons.

**Reply by standenman** [reply]
Thank you!
